pkg/operator/ceph/disruption/machinedisruption/reconcile.go

```diff
@@ -0,0 +1,167 @@
+	r.log.Infof("reconciling %s", request.NamespacedName)
+
+	// Fetching the cephCluster
+	cephClusterInstance := &cephv1.CephCluster{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, cephClusterInstance)
+	if err != nil {
+		r.log.Errorf("%+v", err)
+		return reconcile.Result{}, err
+	}
+
+	// skipping the reconcile since the feature is switched off
+	if !cephClusterInstance.Spec.DisruptionManagement.ManageMachineDisruptionBudgets {
+		r.log.Infof("Skipping reconcile for cephCluster %s as manageMachineDisruption is turned off", request.Name)
+		return reconcile.Result{}, nil
+	}
+
+	mdb := &healthchecking.MachineDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateMDBInstanceName(request.Name, request.Namespace),
+			Namespace: cephClusterInstance.Spec.DisruptionManagement.MachineDisruptionBudgetNamespace,
+		},
+	}
+
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: mdb.GetName(), Namespace: mdb.GetNamespace()}, mdb)
+	if errors.IsNotFound(err) {
+		// If the MDB is not found creating the MDB for the cephCluster
+		maxUnavailable := int32(defaultMaxUnavailable)
+		// Generating the MDB instance for the cephCluster
+		newMDB := &healthchecking.MachineDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateMDBInstanceName(request.Name, request.Namespace),
+				Namespace: cephClusterInstance.Spec.DisruptionManagement.MachineDisruptionBudgetNamespace,
+				Labels: map[string]string{
+					MDBCephClusterNamespaceLabelKey: request.Namespace,
+					MDBCephClusterNameLabelKey:      request.Name,
+				},
+				OwnerReferences: []metav1.OwnerReference{cephCluster.ClusterOwnerRef(cephClusterInstance.GetName(), string(cephClusterInstance.GetUID()))},
+			},
+			Spec: healthchecking.MachineDisruptionBudgetSpec{
+				MaxUnavailable: &maxUnavailable,
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						MDBCephClusterNameLabelKey:      request.Name,
+						MDBCephClusterNamespaceLabelKey: request.Namespace,
+					},
+				},
+			},
+		}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: newMDB.GetName(), Namespace: newMDB.GetNamespace()}, newMDB)
+		if errors.IsNotFound(err) {
+			err = r.client.Create(context.TODO(), newMDB)
+			if err != nil {
+				r.log.Errorf("%+v", err)
+				return reconcile.Result{Requeue: true, RequeueAfter: requeueAfterOneMinute}, err
+			}
+			return reconcile.Result{}, nil
+		} else if err != nil {
+			return reconcile.Result{Requeue: true, RequeueAfter: requeueAfterOneMinute}, err
+		}
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		r.log.Errorf("%+v", err)
+		return reconcile.Result{Requeue: true, RequeueAfter: requeueAfterOneMinute}, err
+	}
+
+	_, isClean, err := cephClient.IsClusterClean(r.context.ClusterdContext, request.Name)
+	if err != nil {
+		r.log.Errorf("%+v", err)
+		maxUnavailable := int32(zeroMaxUnavailable)
```

Comment:
If you're spelling out the value in the variable name, a literal is better than a const.

Reply:
makes sense